app/src/main/java/de/tum/in/tca/ticketcheck/utils/RSASigner.java

```diff
@@ -0,0 +1,75 @@
+package de.tum.in.tca.ticketcheck.utils;
+
+import android.util.Base64;
+
+import com.google.common.base.Charsets;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.SignatureException;
+
+/**
+ * Class providing an API to generate signatures of strings.
+ * Takes care of handling all unicode juggling and crypto algorithm selection.
+ */
+public class RSASigner {
+    /**
+     * A {@link PrivateKey} instance which will be used to generate the signature.
+     */
+    private final PrivateKey privateKey;
+
+    public RSASigner(PrivateKey privateKey) {
+        this.privateKey = privateKey;
+    }
+
+    public static Signature getSignatureInstance() {
+        String signature = "SHA1WithRSA";
```

Comment:
Can we use something else? Like SHA256 maybe?

Reply:
sure